<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1111</width>
    <height>682</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MNIST Testbed</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_genImage">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>301</width>
      <height>301</height>
     </rect>
    </property>
    <property name="title">
     <string>Generated image</string>
    </property>
    <widget class="QLabel" name="label_genImage">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>281</width>
       <height>261</height>
      </rect>
     </property>
     <property name="text">
      <string>Result image</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_modelBuilder">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>10</y>
      <width>771</width>
      <height>441</height>
     </rect>
    </property>
    <property name="title">
     <string>Model Builder</string>
    </property>
    <widget class="QGroupBox" name="groupBox_library">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>181</width>
       <height>101</height>
      </rect>
     </property>
     <property name="title">
      <string>Library</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>169</width>
        <height>61</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QRadioButton" name="radioButton_tensorflow">
         <property name="text">
          <string>TensorFlow</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="radioButton_pytorch">
         <property name="text">
          <string>PyTorch</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="checkBox_gpuMode">
         <property name="text">
          <string>use GPU</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_input">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>20</y>
       <width>201</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>Input / Output</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>12</x>
        <y>30</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_input">
       <item row="1" column="1">
        <widget class="QComboBox" name="comboBox_dataset"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_dataset">
         <property name="text">
          <string>Dataset</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_modelName">
         <property name="text">
          <string>Model name</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="comboBox_model"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_batchSize">
         <property name="text">
          <string>Batch size</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSpinBox" name="spinBox_batchSize">
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_hyperParam">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>391</width>
       <height>241</height>
      </rect>
     </property>
     <property name="title">
      <string>Hyper Parameters</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>31</y>
        <width>371</width>
        <height>201</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_actFnDisc">
         <item>
          <widget class="QCheckBox" name="checkBox_actFnDisc">
           <property name="text">
            <string>Activation for discriminator</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_actFnDisc"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_actFnGen">
         <item>
          <widget class="QCheckBox" name="checkBox_actFnGen">
           <property name="text">
            <string>Activation for generator</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_actFnGen"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_initFn">
         <item>
          <widget class="QCheckBox" name="checkBox_initFn">
           <property name="text">
            <string>Weight initializer</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_initFn"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_hyperParam">
         <item>
          <widget class="QLabel" name="label_hyperParam1">
           <property name="text">
            <string>Hyper parameter 1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_hyperParam1">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.700000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_hyperParam2">
           <property name="text">
            <string>Hyper parameter 2</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBox_hyperParam2">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.700000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_batchNorm">
         <item>
          <widget class="QCheckBox" name="checkBox_bathNorm">
           <property name="text">
            <string>Batch normalization</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QPushButton" name="pushButton_buildModel">
     <property name="geometry">
      <rect>
       <x>550</x>
       <y>400</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Build model</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clearModel">
     <property name="geometry">
      <rect>
       <x>660</x>
       <y>400</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear model</string>
     </property>
    </widget>
    <widget class="QScrollArea" name="scrollArea_modelViewer">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>30</y>
       <width>331</width>
       <height>361</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="palette">
      <palette>
       <active>
        <colorrole role="WindowText">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Dark">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>159</red>
           <green>157</green>
           <blue>154</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Text">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="ButtonText">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
       </active>
       <inactive>
        <colorrole role="WindowText">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Dark">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>159</red>
           <green>157</green>
           <blue>154</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Text">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="ButtonText">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </colorrole>
       </inactive>
       <disabled>
        <colorrole role="WindowText">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>159</red>
           <green>157</green>
           <blue>154</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Dark">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>159</red>
           <green>157</green>
           <blue>154</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="Text">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>159</red>
           <green>157</green>
           <blue>154</blue>
          </color>
         </brush>
        </colorrole>
        <colorrole role="ButtonText">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>159</red>
           <green>157</green>
           <blue>154</blue>
          </color>
         </brush>
        </colorrole>
       </disabled>
      </palette>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::AdjustIgnored</enum>
     </property>
     <property name="widgetResizable">
      <bool>false</bool>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>217</width>
        <height>407</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_training">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>460</y>
      <width>381</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>Training</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>30</y>
       <width>241</width>
       <height>151</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_training">
      <item row="0" column="0">
       <widget class="QLabel" name="label_optimizer">
        <property name="text">
         <string>Optimizer</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBox_optimizer"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_numEpochs">
        <property name="text">
         <string># of Epochs</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinBox_numEpochs">
        <property name="maximum">
         <number>100000</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_learningRateD">
        <property name="text">
         <string>Learning rate for G</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="lineEdit_learningRateG">
        <property name="text">
         <string>0.0002</string>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_learningRateG">
        <property name="text">
         <string>Learning rate for D</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="lineEdit_learningRateD">
        <property name="text">
         <string>0.0002</string>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="pushButton_trainModel">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>30</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Train model</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_evaluation">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>70</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Evaluation</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_tensorBoard">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>110</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>TensorBoard</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_Exit">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>150</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Exit</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_plotPrediction">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>330</y>
      <width>301</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>GAN losses</string>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>29</y>
       <width>281</width>
       <height>181</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_plotPrediction"/>
    </widget>
   </widget>
   <widget class="QScrollArea" name="scrollArea_log">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>470</y>
      <width>341</width>
      <height>181</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_log">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>339</width>
       <height>179</height>
      </rect>
     </property>
     <widget class="QTextEdit" name="textEdit_log">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>341</width>
        <height>181</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
